### PR TITLE
ARN-2036 Amend /assessment/associate endpoint to accept multiple pairs of OASys PKs

### PR DESCRIPTION
**Pull request by devkokov**
Additionally:

1. Use a real Postgres DB for integration tests, as we stand up the full service anyway
2. Provision test data dynamically, in code
3. Removed /assessment/create endpoint, due to overlap with /assessment/associate
4. Removed /assessment/{assessmentUuid}/version/{tag}/answers endpoint due to overlap with /assessment/{assessmentUuid} and the need to surface oasys_equivalent to the UI
5. Re-use the last assessment version if it was created today

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt [hunk 1]
```diff
@@ -18,30 +20,23 @@ class AssessmentVersionService(
   val assessmentVersionRepository: AssessmentVersionRepository,
   val dataMappingService: DataMappingService,
 ) {
-  fun create(tag: String, assessment: Assessment): AssessmentVersion {
+  fun sameOrCloneFromPrevious(tag: String, assessment: Assessment): AssessmentVersion {
```
Comment (LucasCairns) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt:23]: we could retain the `create` verb here for example `createOrCloneFromPrevious`

Comment (devkokov) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt:23]: 🤔 clone and create are the same thing in this context... maybe "createOrReuse"?

Comment (devkokov) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt:23]: or "createOrGetPrevious"?

**Review by LucasCairns — APPROVED** [review]
LGTM! 👍  Nice work, I like the move to Postgres as a test database, we've had issues in the past that only surface in Postgres/H2 resulting in failed deployments, so being consistent here is always a good thing

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt (hunk 1 above)
Comment (LucasCairns) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt:23]: true and makes sense, maybe `createNewOrCloneFromPrevious`, but I'm also happy with either of the above suggestions 👍

Comment (devkokov) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt:23]: ah I see what you mean now. There's actually 3 scenarios here:
- create new (no previous version exists)
- clone from previous (previous version is older than 1 day)
- get previous (previous version was created today)

so I guess `createNewOrGetPreviousOrCloneFromPrevious`

Comment (devkokov) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsstrengthsbasedneedsassessmentsapi/service/AssessmentVersionService.kt]: Think I've settled on `getPreviousOrCreate` :D describes all scenarios while not being ridiculously long

**Review by LucasCairns — APPROVED** [review]
LGTM 👍

**Pull request merged**